src/software/simulation/simulator.cpp

```diff
@@ -98,7 +123,30 @@ void Simulator::setPrimitives(ConstPrimitiveVectorPtr primitives)
     }
 }
 
-World Simulator::getWorld()
+void Simulator::stepSimulation(const Duration& time_step)
+{
+    SimulatorBallSingleton::setSimulatorBall(simulator_ball);
```

Comment:
what is this call needed for?

Reply:
This sets the ball being used in each `firmware_world`. We only do this once at the top since all robots "see" and interact with the same ball. Added a comment to help clarify.